### PR TITLE
Use 409 error code for duplicate issuer when adding km

### PR DESCRIPTION
**Pull request by menakaj**
Fixes https://github.com/wso2-enterprise/choreo/issues/9642

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java [hunk 1]
```diff
@@ -521,6 +523,30 @@ public KeyManagerConfigurationDTO addKeyManagerConfiguration(
...
+                if (idp != null) {
+                    throw new APIManagementException(
+                            String.format("Key Manager with issuer %s is already registered in tenant %s",
+                                    keyManagerConfigurationDTO.getAdditionalProperties().get(ISSUER_KEY).toString(),
+                                    keyManagerConfigurationDTO.getOrganization()), ExceptionCodes.DUPLICATE_ISSUER);
+                }
+            } catch (IdentityProviderManagementException e) {
+                throw new APIManagementException(String.format("IdP adding failed. %s", e.getMessage()),
+                        e,
+                        ExceptionCodes.IDP_ADDING_FAILED);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java:543]: This is most probably an Internal Server Error right. But IDP_ADDING_FAILED returning 400 may not give the correct information. Do we know what other places IDP_ADDING_FAILED is used and based on that is it okay to change its status code to 500?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java [hunk 2]
```diff
@@ -521,6 +523,30 @@ public KeyManagerConfigurationDTO addKeyManagerConfiguration(
...
+            }
+        } else {
+            throw new APIManagementException("IdP adding failed. The issuer cannot be empty.",
+                    ExceptionCodes.IDP_ADDING_FAILED);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java:547]: Better to have a new Exception Code with "The issuer cannot be empty" as the description.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java [hunk 3]
```diff
@@ -521,6 +523,30 @@ public KeyManagerConfigurationDTO addKeyManagerConfiguration(
...
+        if (keyManagerConfigurationDTO.getAdditionalProperties().get(ISSUER_KEY) != null) {
+            try {
+                IdentityProvider idp = IdentityProviderManager.getInstance().getIdPByMetadataProperty(
+                        IDP_ISSUER_NAME,
+                        keyManagerConfigurationDTO.getAdditionalProperties().get(ISSUER_KEY).toString(),
+                        APIUtil.getInternalOrganizationDomain(keyManagerConfigurationDTO.getOrganization()),
+                        false);
+                if (idp != null) {
+                    throw new APIManagementException(
+                            String.format("Key Manager with issuer %s is already registered in tenant %s",
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java:536]:
```suggestion
                            String.format("Key Manager with issuer %s is already registered in organization %s",
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java (hunk 1 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java:543]: IDP_ADDING_FAILED is used in two places. 

1. Add IDP https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java#L516-L543
2. Edit IDP https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java#L677-L712

Both are thrown catching the IdentityProviderManagementException. I think both scenarios have similar behavior.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java (hunk 2 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java:547]: ACK

Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/11118/commits/c7402c0dac7b184bd56abb1513f4cad797ce0917

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java (hunk 1 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java]: Changed the error codes with https://github.com/wso2/carbon-apimgt/pull/11118/commits/c7402c0dac7b184bd56abb1513f4cad797ce0917

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**